src/components/Prompts/PromptPanel/PromptPanel.tsx

```diff
@@ -72,22 +74,55 @@ export const PromptPanel: FC<PromptPanelProps> = (props) => {
     }
   };
 
+  const processOutcomeUrl = async () => {
+    setLoading(true);
+    let jsonPayload = {
+      answers: gameInfoContext.answers,
+      avatarId: gameInfoContext.avatar?.id,
+      personaId: gameInfoContext.persona?.id,
+      themeId: gameInfoContext.theme?.id,
+    };
+
+    // Check if JSON payload (answers, etc) already exists and if it does redirect
+    const entityResult = await AzureProxyService().getByJson(JSON.stringify(jsonPayload));
+
+    if (entityResult?.result?.rowKey) {
+      return `/outcome/${entityResult.result.rowKey}`;
+    }
+
+    // JSON Payload doesn't exist so lets create it
+    const rowKey = uuidv4();
+    const postPayload = {
+      rowKey,
+      json: jsonPayload,
+    };
+
+    const createResult = await AzureProxyService().createEntity(JSON.stringify(postPayload));
+
+    if (createResult.success) {
+      return `/outcome/${rowKey}`;
+    }
+
+    setLoading(false);
+  };
+
   const triggerNextPrompt = async () => {
-    // Next Prompt is based on Pool of Questions that are not answered yet, Collection is FIFO (First In First Out)
-    if (gameInfoContext.questionsBank?.get() !== undefined) {
-      if (gameInfoContext.questionsBank.get()!.length > 0) {
-        const questionQueue = gameInfoContext.questionsBank.get();
-        const nextPrompt = questionQueue!.shift();
-        gameInfoContext.questionsBank.set(questionQueue);
+    const questionBank = gameInfoContext.questionsBank.get();
 
-        setCurrentPrompt(nextPrompt);
+    if (questionBank !== undefined && questionBank.length > 0) {
+      const questionQueue = gameInfoContext.questionsBank.get();
+      const nextPrompt = questionQueue!.shift();
+      gameInfoContext.questionsBank.set(questionQueue);
 
-        await trackPromptPageView(nextPrompt);
-      } else {
-        router.push('/outcome');
-      }
+      setCurrentPrompt(nextPrompt);
+
+      await trackPromptPageView(nextPrompt);
     } else {
-      router.push('/outcome');
+      const urlString = await processOutcomeUrl();
+
+      if (urlString) {
```

Comment:
processOutcomeUrl may not return a urlString. What should we do in that case? Right now it looks like we do not navigate and leave user stranded on the prompt page.

Reply:
This is mentioned in the 404 comment above, but I could push the user to a /outcome/error page or something like that and it would behave exactly like the 404 page because error wouldn't be a valid outcome.  And then I could configure the outcome to instead of loading the outcome from CH1 etc, it would load a standalone description about how that users outcome doesn't match their inputs or something along those lines and provide a button to start over.

Thoughts?

Reply:
I like the error page idea.

Reply:
Okay great, I'll proceed with this approach.